src/material/radio/radio.ts

```diff
@@ -489,13 +490,18 @@ export abstract class _MatRadioButtonBase extends _MatRadioButtonMixinBase imple
               private _focusMonitor: FocusMonitor,
               private _radioDispatcher: UniqueSelectionDispatcher,
               public _animationMode?: string,
-              private _providerOverride?: MatRadioDefaultOptions) {
+              private _providerOverride?: MatRadioDefaultOptions,
+              tabIndex?: string) {
     super(elementRef);
 
     // Assertions. Ideally these should be stripped out by the compiler.
     // TODO(jelbourn): Assert that there's no name binding AND a parent radio group.
     this.radioGroup = radioGroup;
 
+    if (tabIndex) {
+      this.tabIndex = parseInt(tabIndex) || 0;
```

Comment:
Prefer `Number(tabIndex)` to `parseInt`

Reply:
I ended up using `coerceNumberProperty` which has `Number` internally with some fallback logic.